### PR TITLE
llmq: Verify final commitments in CDKGSession::FinalizeCommitments

### PR DESCRIPTION
**Pull request by xdustinface**
Based on #4032

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
utACK

File: src/llmq/quorums_dkgsession.cpp [hunk 1]
```diff
@@ -1278,6 +1278,12 @@ std::vector<CFinalCommitment> CDKGSession::FinalizeCommitments()
             logger.Batch("failed to recover quorum sig");
             continue;
         }
+
+        if (!fqc.Verify(pindexQuorum, true)) {
+            logger.Batch("failed to verify final commitment");
+            continue;
+        }
+
```
Comment (UdjinM6) [inline src/llmq/quorums_dkgsession.cpp:1286]: Let's move this below `t2.stop();`, wrap in its own timer and tweak the log below accordingly

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
Changed my mind 🙈

File: src/llmq/quorums_dkgsession.cpp (hunk 1 above)
Comment (xdustinface) [reply src/llmq/quorums_dkgsession.cpp]: Makes sense 👍 See 7621990

**Review by UdjinM6 — APPROVED** [review]
👍 

utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK

**Pull request merged**